src/fileActions.jsx

```diff
@@ -266,8 +266,15 @@ const EditPermissionsModal = ({ selected, path }) => {
     const [mode, setMode] = useState(selected.mode);
     const [group, setGroup] = useState(selected.group);
     const [errorMessage, setErrorMessage] = useState(undefined);
-    const accounts = useFile("/etc/passwd", { syntax: etcPasswdSyntax });
-    const groups = useFile("/etc/group", { syntax: etcGroupSyntax });
+    const [accounts, setAccounts] = useState(null);
+    const [groups, setGroups] = useState(null);
+
+    useInit(async () => {
+        const passwd = await cockpit.spawn(["getent", "passwd"], { err: "message" });
+        setAccounts(etcPasswdSyntax.parse(passwd));
+        const group = await cockpit.spawn(["getent", "group"], { err: "message" });
+        setGroups(etcGroupSyntax.parse(group));
```

Comment:
Err, wait -- this should handle exceptions to prevent a page crash. `useFile()` would log errors to the console, so a `catch(ex)` should do the same here.